projects/gloo/cli/pkg/cmd/gatewayapi/convert/metrics.go

```diff
@@ -3,119 +3,58 @@ package convert
-	gatewayAPIConfigMetrics.WithLabelValues("Gateway").Add(float64(len(output.gatewayAPICache.Gateways)))
-	gatewayAPIConfigMetrics.WithLabelValues("AuthConfig").Add(float64(len(output.gatewayAPICache.AuthConfigs)))
-	gatewayAPIConfigMetrics.WithLabelValues("HTTPRoute").Add(float64(len(output.gatewayAPICache.HTTPRoutes)))
-	gatewayAPIConfigMetrics.WithLabelValues("Upstream").Add(float64(len(output.gatewayAPICache.Upstreams)))
-	gatewayAPIConfigMetrics.WithLabelValues("RouteOption").Add(float64(len(output.gatewayAPICache.RouteOptions)))
-	gatewayAPIConfigMetrics.WithLabelValues("VirtualHostOption").Add(float64(len(output.gatewayAPICache.VirtualHostOptions)))
-	gatewayAPIConfigMetrics.WithLabelValues("ListenerSets").Add(float64(len(output.gatewayAPICache.ListenerSets)))
-	gatewayAPIConfigMetrics.WithLabelValues("HTTPListenerOption").Add(float64(len(output.gatewayAPICache.HTTPListenerOptions)))
-	gatewayAPIConfigMetrics.WithLabelValues("DirectResponse").Add(float64(len(output.gatewayAPICache.DirectResponses)))
-	gatewayAPIConfigMetrics.WithLabelValues("ListenerOption").Add(float64(len(output.gatewayAPICache.ListenerOptions)))
-	gatewayAPIConfigMetrics.WithLabelValues("Unknown").Add(float64(len(output.gatewayAPICache.YamlObjects)))
-
-	metrics, _ := prometheus.DefaultGatherer.Gather()
-	_, err := fmt.Fprintf(os.Stdout, "-------------------------------------\n")
-	if err != nil {
-		return
-	}
-	for _, m := range metrics {
-		if m.GetName() == "gloo_config_count" {
-			var count float64
-			for _, t := range m.GetMetric() {
-				if t.GetLabel()[0] != nil && t.GetCounter() != nil {
-					_, _ = fmt.Fprintf(os.Stdout, "Gloo Config: Number of %s: %v\n", t.GetLabel()[0].GetValue(), t.GetCounter().GetValue()-1)
-					if t.GetLabel()[0].GetValue() != "Unknown" {
-						count += t.GetCounter().GetValue() - 1
-					}
-				}
-			}
-			_, _ = fmt.Fprintf(os.Stdout, "Total Gloo Config: %v\n", count)
-		}
-	}
-	_, err = fmt.Fprintf(os.Stdout, "-------------------------------------\n")
-	if err != nil {
-		return
-	}
-	for _, m := range metrics {
-		if m.GetName() == "gatewayapi_config_count" {
-			var count float64
-			for _, t := range m.GetMetric() {
-				if t.GetLabel()[0] != nil && t.GetCounter() != nil {
-					_, _ = fmt.Fprintf(os.Stdout, "Gateway API Config: Number of %s: %v\n", t.GetLabel()[0].GetValue(), t.GetCounter().GetValue()-1)
-					if t.GetLabel()[0].GetValue() != "Unknown" {
-						count += t.GetCounter().GetValue() - 1
-					}
-				}
-			}
-			_, _ = fmt.Fprintf(os.Stdout, "Total Gateway API Config: %v\n", count)
-		}
-	}
-	_, err = fmt.Fprintf(os.Stdout, "-------------------------------------\n")
-	if err != nil {
-		return
-	}
-	for _, m := range metrics {
-		if m.GetName() == "files_evaluated" {
-			if m.GetMetric()[0] != nil && m.GetMetric()[0].GetCounter() != nil {
-				_, _ = fmt.Fprintf(os.Stdout, "Files evaluated: %v\n", m.GetMetric()[0].GetCounter().GetValue())
-			}
-		}
+func printMetrics(output *GatewayAPIOutput, filesEvaluated int) {
```

Comment:
nit: Sometimes Ill actually invert this method and instead of accepting a gatewayAPIObjet, put a method on the GatewayAPIObject that either returns a string representation of it, or a "Write" methods taht accepts a writer and writes the contents to a writer. That way as the shape of the GatewayAPIObject changes, the method would get updated as part of it, and not forgotten about